CMakeLists.txt

```diff
@@ -61,16 +61,15 @@ endif ()
 
 if ("${CMAKE_CXX_COMPILER_ID}" MATCHES ".*Clang") # both Clang and AppleClang
   set (is_clang TRUE)
-  # TODO enable this version check
   if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" AND
-         CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.0)
-    message (FATAL_ERROR "This project requires clang 5 or later")
+         CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0)
+    message (FATAL_ERROR "This project requires clang 7 or later")
```

Comment:
I'm a little conflicted about this check. The problem with clang 5 is the interaction with gcc (as you know). I think it's reasonable to say "we only support clang 7 or newer". However, this is refusing to build with older clangs, even if they would work.

What do you think about making this a warning? Keeping it as an error is also reasonable. I suspect there are very few people building with clang 5 anyway.

Reply:
I'm fine keeping it at 5, but I think we need the travis builds to cover it then...otherwise it's untested. The 5 and 6 builds work just fine in travis, so if we want 5 as min ver, then we add the builds back in...I think you agree?